File: src/main/java/de/burrotinto/burroPlayer/adapter/status/GPIOStatusAdapter.java
package de.burrotinto.burroPlayer.adapter.status;

import de.burrotinto.burroPlayer.port.gpio.GPIOFacade;
import de.burrotinto.burroPlayer.values.PinValue;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

/**
 * Created by derduke on 27.03.17.
 */
@Service
@RequiredArgsConstructor
public class GPIOStatusAdapter implements StatusAdapter {
    public final static long BLINK_DURATION = 10;
    private final GPIOFacade gpioFacade;
    private final PinValue pin;

    @Override
    public void somethingHappens() {
        gpioFacade.blink(pin.getHappening(), BLINK_DURATION);
    }
}

File: src/main/kotlin/de/burrotinto/burroplayer/adapter/status/GPIOStatusAdapter.kt
package de.burrotinto.burroplayer.adapter.status

import de.burrotinto.burroplayer.port.gpio.GPIOFacade
import de.burrotinto.burroplayer.values.PinValue
import lombok.RequiredArgsConstructor
import org.springframework.stereotype.Service

/**
 * Created by derduke on 27.03.17.
 */
@Service
@RequiredArgsConstructor
class GPIOStatusAdapter(private val gpioFacade: GPIOFacade,
                        private val pin: PinValue) : StatusAdapter {


    override fun somethingHappens() {
        gpioFacade.blink(pin.happening, BLINK_DURATION)
    }

    companion object {
        val BLINK_DURATION: Long = 10
    }
}
